src/mlpack/core/optimizers/cmaes/cmaes.hpp

```diff
@@ -0,0 +1,398 @@
+/**
+ * @file cmaes.h
+ * @author Kartik Nighania (GSoC 17 mentor Marcus Edel)
+ *
+ * Covariance Matrix Adaptation Evolution Strategy
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+#define MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+
+#include <cmath>
+#include <limits>
+#include <ostream>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <iostream>
+#include <armadillo>
+#include <iostream>
+
+#include "random.hpp"
+
+namespace mlpack {
+namespace optimization {
+//vector use of pop
+//cmaes as init
+/**
+ * @class CMAES 
+ * all the function parameters available to the user
+ */
+template<typename funcType, typename T>
```

Comment:
yes @zoq ..  the user can choose its own type of function type and the algorithm working data type independently depending on its application. Now i think just working with double is fine..